### PR TITLE
take timestamp from arduino

### PR DESCRIPTION
**Pull request by mwittgen**

### CODE REVIEW

**Comment by slacrherbst** [comment]
Looks generally ok, but does not address a problem I considered when thinking about this. The 16-bit millis value will rollover quite frequently. I think we need to keep a local counter with infinite depth and adjust that value by the difference between the current millis() value and the previous millis() value. 

localTime = lastLocal + (currMillis - lastMillis)

but we have to deal with the rollover condition and deal with that appropriately. 

if currMillis < lastMillis: some 16-bit adjust subtraction here.

File: python_client/ambu_control.py [hunk 1]
```diff
@@ -160,14 +161,19 @@ def _handleSerial(self):
...
+                    count  = int(data[2],0)
+                    press  = float(data[3])
+                    flow   = float(data[4])
+                    vol    = float(data[5])
+                    #diffT  = ts - self._stime
+                    if self._smillis==-1: 
+                        self._smillis=millis
+                        continue
+                    else:
+                        diffT=(millis-self._smillis)/1000.
```
Comment (slacrherbst) [inline python_client/ambu_control.py:176]: I don't this this is enough. this may result in a negative number.

**Comment by slacrherbst** [comment]
Ok. I agree you are correct. I had it in my head from previous readings that millis() was 16-bit. I am wrong.

File: python_client/ambu_control.py (hunk 1 above)
Comment (mwittgen) [reply python_client/ambu_control.py:176]: Noted: it should have some roll-over handling here. In principle just discarding a negative interval every 50 days is probably fine.

**Pull request merged**